From 0643b985787991e84c257beaf823da1e53e14755 Mon Sep 17 00:00:00 2001
From: Russell Bryant <rbryant@redhat.com>
Date: Fri, 30 Nov 2012 16:08:52 -0500
Subject: [PATCH] Fix rpc control_exchange regression.

A change was made in rpc that requires project's to define the
control_exchange option themselves.  This was so the project's could
set their own default value.  This change has made it in to nova without
adding the control_exchange option, meaning that it is using the default
of 'openstack', potentially conflicting with other projects.

See https://review.openstack.org/#/c/12518/

Fix bug 1083944.

Change-Id: Ifc00a537c4ee327cbee91fe1d911f8c402e350b7
---
 nova/flags.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/nova/flags.py b/nova/flags.py
index 20ae652..7b8c841 100644
--- a/nova/flags.py
+++ b/nova/flags.py
@@ -425,6 +425,9 @@ global_opts = [
                    'include: libvirt.LibvirtDriver, xenapi.XenAPIDriver, '
                    'fake.FakeDriver, baremetal.BareMetalDriver, '
                    'vmwareapi.VMWareESXDriver'),
+    cfg.StrOpt('control_exchange',
+               default='nova',
+               help='AMQP exchange to connect to if using RabbitMQ or Qpid'),
 ]
 
 FLAGS.register_opts(global_opts)
